src/main/java/org/jitsi/jicofo/JitsiMeetConferenceImpl.java

```diff
@@ -734,6 +737,62 @@ private ColibriConference createNewColibriConference(Jid bridgeJid)
     }
 
     /**
+     * Creates a new {@link ColibriConference} and creates a {@link JvbApi}
+     * instance for it to use to control the JVB
+     * @param bridgeJid see
+     *                  {@link JitsiMeetConferenceImpl#createNewColibriConference(Jid)}
+     * @param jvbApiUrl the base URL of the JVB API instance
+     * @param jvbApiPort the port that the JVB API is listening on
+     * @return the created {@link ColibriConference}
+     * @throws XmppStringprepException see
+     * {@link JitsiMeetConferenceImpl#createNewColibriConference(Jid)}
+     */
+    private ColibriConference createNewColibriConference(Bridge bridge)
+        throws XmppStringprepException
+    {
+        ColibriConference conf =  createNewColibriConference(bridge.getJid());
+
+        if (bridge.getJvbApiUrl() != null &&
+            bridge.getJvbApiPort() != null &&
+            bridge.getSupportedApiVersions() != null)
+        {
+            logger.info("Bridge advertised a JVB API address.  JVB API " +
+                "client supports " + VersionKt.SUPPORTED_API_VERSIONS +
+                ", bridge supports " + bridge.getSupportedApiVersions());
+            ApiVersion maxApiVersion =
+                VersionKt.SUPPORTED_API_VERSIONS.maxSupported(
+                    bridge.getSupportedApiVersions());
+            if (maxApiVersion == null)
+            {
+                logger.info("No JVB API version compatibility found " +
+                    "between client and server");
+            }
+            else
+            {
+                logger.info("Got common api version " + maxApiVersion);
+                switch (maxApiVersion)
+                {
+                    case V1:
+                    {
+                        conf.setJvbApi(
+                            new org.jitsi.videobridge.api.client.v1.JvbApi(
+                                bridge.getJvbApiUrl(),
+                                bridge.getJvbApiPort()
+                            )
+                        );
+                        break;
+                    }
+                    default: throw new RuntimeException("Found common JVB " +
```

Comment:
If the api dependency is bumped such that it implements a new version, but this code isn't updated, won't this fail even if both client and server still support v1?

Reply:
`maxApiVersion` could probably be named better: it's the calculated `max common supported API version` between the JVB and Jicofo (see line 763).

Reply:
Yeah, but it's the max version supported by the JVB API module being used by this code, which isn't necessarily the max version supported by the code locally here in jicofo, isn't it?

(I.e., consider that the JVB API is updated to support v2 and the version is bumped in the jicofo pom, but the jicofo code proper isn't updated.)

Reply:
There are 2 pieces:
The version(s) supported by the `jvb-api-client` version that Jicofo is using, and the versions supported by the `jvb-api-server` (in this case, the JVB itself).  The 'server' versions are advertised by the JVB in its presence.  When JVB supports v2, it will advertise support for both v1 and v2, so we'd still settle on v1 here.

Reply:
Talked offline and I see the case jonathan is referring to...pushed an update to take into account which API versions Jicofo has _implemented_ support for as well.

Reply:
Did you want to actually add the code which uses the intersect?

Reply:
It's in there...line 773